src/Integration.Vsix/SonarLintTagger/TextBufferIssueTracker.cs

```diff
@@ -274,14 +274,26 @@ void IIssueConsumer.Accept(string path, IEnumerable<Issue> issues)
                 return;
             }
 
-            // TODO: collect the additional information required to compare be able to filter
-            // the issues (i.e. the text of the line and the line hash)
-            // issues = issuesFilter.Filter(path, issues);
+            var filteredIssues = FilterIssues(path, issues);
 
-            var newMarkers = issues.Where(IsValidIssueTextRange).Select(CreateIssueMarker);
+            var newMarkers = filteredIssues.Where(IsValidIssueTextRange).Select(CreateIssueMarker);
             UpdateIssues(newMarkers);
         }
 
+        private IEnumerable<Issue> FilterIssues(string path, IEnumerable<Issue> issues)
+        {
+            var filterableIssues = IssueToFilterableIssueConverter.Convert(issues, currentSnapshot);
+
+            var filteredIssues = issuesFilter.Filter(path, filterableIssues);
```

Comment:
I realize now that we have a bit of naming problem:
1. The method is called FilterIssues, but IsValidIssueTextRange is not part of it
2. The class is called issuesFilter, so presumably we don't know what's the criteria, but then we write to log suppressedCount.

If we keep it "issuesFilter", then I think the  suppression log should be written inside the issuesFilter class. And maybe we should call IsValidIssueTextRange before filtering suppressions, since they are not valid anyway. Wdyt?

Reply:
I'll rename the method to `RemoveSuppressedIssues`.